### ISSUE TITLE
device default.mixed &&  gradient disappear

### ISSUE THREAD

**Issue by gtezg30062** [PennyLaneAI/pennylane#7102]
Hi, after changing the device from **default.qubit** to **default.mixed**, the first batch of quantum-classical hybrid neural network training had a vanishing gradient problem (it can be trained normally when using device default.qubit). How can this problem be solved?

**ValueError**: State vectors have to be of norm 1.0, vector 0 has squared norm nan.

**Comment by albi3ro** [comment]
Hi @gtezg30062 and thanks for opening this issue.

Can provide a minimal non-working example so we can replicate the issue?

**Comment by gtezg30062** [comment]
Thanks for your reply. The minimal non-working example is in the link.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PennyLaneAI/pennylane"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年3月18日(星期二) 晚上9:19
***@***.***&gt;;
抄送:&nbsp;"琴笙  阡陌  ***@***.******@***.***&gt;;
主题:&nbsp;Re: [PennyLaneAI/pennylane] device default.mixed &amp;&amp;  gradient disappear (Issue #7102)



   
Hi @gtezg30062 and thanks for opening this issue.
 
Can provide a minimal non-working example so we can replicate the issue?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***&gt;
  albi3ro left a comment (PennyLaneAI/pennylane#7102)
 
Hi @gtezg30062 and thanks for opening this issue.
 
Can provide a minimal non-working example so we can replicate the issue?

**Comment by CatalinaAlbornoz** [comment]
Hi @gtezg30062 ,

The error doesn't seem to indicate vanishing gradients. It looks like something else might be happening there. Have you used `default.mixed` before? If not then I recommend that you start by reading our demo on [Noisy circuits](https://pennylane.ai/qml/demos/tutorial_noisy_circuits) and the Codebook module on [Noisy Quantum Theory](https://pennylane.ai/codebook/noisy-quantum-theory). These can help you learn about density matrices, which might be the reason why you're getting an unexpected output.  

If these resources don't help you solve the issue we ask that you please share a minimal reproducible example so that we can look for the root of the problem. A minimal reproducible example (or minimal working example) is the simplest version of the code that reproduces the problem. It should be self-contained, including all necessary imports, data, functions, etc., so that we can copy-paste the code and reproduce the problem. However it shouldn't contain any unnecessary data, functions, etc. (for example gates and functions that can be removed to simplify the code). Since the link you shared is the link to this same issue, at the moment we can't really see what the issue might be. 

Please also share your full error traceback and the output of `qml.about()`.
I hope these resources help you resolve your issue!

**Comment by gtezg30062** [comment]
@CatalinaAlbornoz Thanks, this helped lot!
